File: BST/ruby/tree.rb
require_relative 'node'

class Tree 
    @root

    def initialize()
    end

    def add(value)
       if @root.nil?
            @root = Node.new(value)
            return true
       end
       return @root.add(value)
    end

    def contain(value)
        if @root.nil?
            return false
        end
        return @root.contain(value)
     end

     def length()
        if @root.nil?
            return 0
        end
        return @root.length()
     end

     def height()
        if @root.nil?
            return 0
        end
        return @root.height()
     end

end

File: BST/julia/tree.jl
include("node.jl")

mutable struct Tree
    root::Union{Node, Nothing}
end

function add(self::Tree,value::Int)
    if isnothing(self.root)
        self.root = Node(value, nothing, nothing)
        return true
    end
    return add(self.root, value)
end

function contain(self::Tree,value::Int)
    if isnothing(self.root)
        return false
    end
    return contain(self.root, value)
end

function length(self::Tree)
    if isnothing(self.root)
        return 0
    end
    return length(self.root)
end
function height(self::Tree)
    if isnothing(self.root)
        return 0
    end
    return height(self.root)
end

